platform/viewer/src/lib/filesToStudies.js

```diff
@@ -1,160 +1,20 @@
-          windowWidth: DICOMWeb.getString(instance['00281051']),
-          rescaleIntercept: DICOMWeb.getNumber(instance['00281052']),
-          rescaleSlope: DICOMWeb.getNumber(instance['00281053']),
-          rescaleType: DICOMWeb.getNumber(instance['00281054']),
-          sourceImageInstanceUid: getSourceImageInstanceUid(instance),
-          laterality: DICOMWeb.getString(instance['00200062']),
-          viewPosition: DICOMWeb.getString(instance['00185101']),
-          acquisitionDateTime: DICOMWeb.getString(instance['0008002A']),
-          frameIncrementPointer: getFrameIncrementPointer(instance['00280009']),
-          frameTime: DICOMWeb.getNumber(instance['00181063']),
-          frameTimeVector: parseFloatArray(
-            DICOMWeb.getString(instance['00181065'])
-          ),
-          sliceThickness: DICOMWeb.getNumber(instance['00180050']),
-          spacingBetweenSlices: DICOMWeb.getString(instance['00180088']),
-          lossyImageCompression: DICOMWeb.getString(instance['00282110']),
-          derivationDescription: DICOMWeb.getString(instance['00282111']),
-          lossyImageCompressionRatio: DICOMWeb.getString(instance['00282112']),
-          lossyImageCompressionMethod: DICOMWeb.getString(instance['00282114']),
-          echoNumber: DICOMWeb.getString(instance['00180086']),
-          contrastBolusAgent: DICOMWeb.getString(instance['00180010']),
-          radiopharmaceuticalInfo: getRadiopharmaceuticalInfo(instance),
-          wadouri: WADOProxy.convertURL(wadouri, server),
-          wadorsuri: WADOProxy.convertURL(wadorsuri, server),*/
-        });
-      });
-    });
-
-    studies.push(study);
-  });
-
-  console.warn(studies);
+const processFile = async file => {
+  const fileLoaderService = new FileLoaderService(file);
+  const imageId = fileLoaderService.addFile(file);
+  const image = await fileLoaderService.loadFile(file, imageId);
+  const datasets = await fileLoaderService.getDataset(image, imageId);
+  const studies = await fileLoaderService.getStudies(datasets, imageId);
 
   return studies;
-}
+};
 
 export default async function filesToStudies(files) {
-  const imagePromises = files.map(file => {
-    const imageId = cornerstoneWADOImageLoader.wadouri.fileManager.add(file);
-    return cornerstone
-      .loadAndCacheImage(imageId)
-      .catch(error => console.warn(error));
-  });
-
-  const images = await Promise.all(imagePromises);
-  const datasets = [];
-
-  images.forEach(image => {
-    if (!image || !image.data) {
-      return;
-    }
-
-    const arrayBuffer = image.data.byteArray.buffer;
-    const dicomData = dcmjs.data.DicomMessage.readFile(arrayBuffer);
-    const dataset = dcmjs.data.DicomMetaDictionary.naturalizeDataset(
-      dicomData.dict
-    );
-    dataset._meta = dcmjs.data.DicomMetaDictionary.namifyDataset(
-      dicomData.meta
-    );
-    dataset.imageId = image.imageId;
-
-    datasets.push(dataset);
+  const processFilesPromises = files.map(file => {
```

Comment:
I think you can just write `files.map(processFile)`

Reply:
yeah.. thank you